libbpf-rs/src/print.rs

```diff
@@ -23,16 +24,31 @@ impl From<libbpf_sys::libbpf_print_level> for PrintLevel {
     }
 }
 
-pub type PrintCallback = fn(PrintLevel, &str);
+pub type PrintCallback = fn(PrintLevel, &dyn fmt::Display);
```

Comment:
I think the simplest solution is expanding the `set_print` API to take in the log level to filter by a parameter, and keeping this print callback as a `fn(PrintLevel, &str)`. One additional benefit of this is that users also then no longer have to do an if check themselves in their callback function to filter logs by the log level they want.

Other than this, your commits LGTM!

Reply:
I'm not going to push back very hard on this, because it's all minor details, but I still don't know about adding a filter parameter:

1. Should the parameter be a minimum level to show or a maximum level to hide?
2. Inclusive or exclusive?
3. Depending on the previous answers, we'll need to introduce a PrintLevel::All or PrintLevel::None that does not exist in libbpf.

I really don't think that requiring the user to write an if-statement is much of a burden, especially because I'm of the opinion that the most common use will be to pass this on to `log` or `syslog` or something, which means users will need a match to convert `PrintLevel` to `log::Level`, etc. anyway..

Reply:
Reflecting on this, maybe the real solution is to make take a `dyn Fn(...) + 'static` so that we can pass closures as well as function pointers. That would be a much nicer interface, and there's really no reason I couldn't have done it that way.

Reply:
I don't think we need to add PrintLevel::All or PrintLevel::None. The way I envision it working is hierarchically -  if the filter is set to LIBBPF_WARN, then only LIBBPF_WARN messages should go through (LIBBPF_INFO and LIBBPF_DEBUG messages should be discarded). If the filter is set to LIBBPF_INFO, then LIBBPF_WARN and LIBBPF_INFO go through while LIBBPF_DEBUG messages are discarded. And if the filter is set to LIBBPF_DEBUG messages, then all messages go through.

Reply:
How should this work with restoring a previous callback?

Reply:
We could add a getter function that will return to the user the current callback and print level, and correspondingly  `set_print` would return back nothing. If the user has a particular callback + print level they want to restore in the future, then the user is responsible for getting and keeping that around, before the user replaces `set_print` with a new callback + print level. What are your thoughts on this?

Reply:
I can implement that. We'll still need a PrintLevel::None if we want to avoid the vsprintf penalty when turning off log messages completely. Or the callback can be an Option.